pkg/agent/plugin/nodeattestor/devid/devid.go

```diff
@@ -0,0 +1,368 @@
+package devid
+
+import (
+	"context"
+	"crypto/x509"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"sync"
+
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/hcl"
+	"github.com/spiffe/spire/pkg/agent/plugin/nodeattestor"
+	"github.com/spiffe/spire/pkg/agent/plugin/nodeattestor/devid/tpm"
+	"github.com/spiffe/spire/pkg/common/catalog"
+	"github.com/spiffe/spire/pkg/common/plugin/devid"
+	spc "github.com/spiffe/spire/proto/spire/common"
+	"github.com/spiffe/spire/proto/spire/common/plugin"
+)
+
+const (
+	// defaultDevicePath is the value used when tpm_device_path is not set
+	defaultDevicePath = "/dev/tpmrm0"
```

Comment:
I like the idea of trying to auto-detect the TPM in case the path is not explicitly configured instead of using a default value.

> How likely is it that the tpm device path is different than this? 

For linux it depends on the kernel version, the TPM version and the number of TPMs: 

- For kernel versions lower than v4.12-rc1, TPMs v1.2 and v2.0 are exposed as `/dev/tpm[n]`
- Starting on kernel version v4.12-rc1, TPM2.0 are exposed as `/dev/tpmrm[n]`   

> Is there any way we could detect the TPM device location in the event that it's not explicitly configured?

Yes, we can do that by looking for `/dev/tpm*`/ `/dev/tpmrm*` patterns and verifying the TPM version to be 2.0.  In case that more than one TPM is exposed, we can take the first one.

What do you think about this approach?